### ISSUE TITLE
A lambda created in a VPC with specific subnets and security groups is associated with subnets and security groups not configured in s-fuction.json

### ISSUE THREAD

**Issue by hypexr** [serverless/serverless#1181]
Serverless v0.5.5

My VPC and subnet configuration is (created in s-resources-cf.json):

VPC 10.0.0.0/16

Subnets
compute subnet 1 (for lambdas) 10.0.0.0/19
compute subnet 2 (for lambdas) 10.0.32.0/19
db subnet 1 10.0.64.0/24
db subnet 2 10.0.65.0/24

Security Groups
analytics-dev-lambda-security-group (for lambdas)
analytics-dev-db-security-group (for DB)

In s-function.json I have a vpc configuration like this.
  "vpc": {
    "securityGroupIds": [ "${lambdaSecurityGroup}" ],
    "subnetIds": [ "${computeSubnet1}", "${computeSubnet2}" ]
  }

When I deploy the version I expect to see these in the Advanced Settings in the AWS console:
Subnets: compute subnet 1, 2
Security groups: analytics-dev-lambda-security-group

But I see all of the subnets and security groups in the VPC listed:

VPC
vpc-*********

Subnets
subnet-********* (10.0.64.0/24) | us-east-1a analytics-dev-db-subnet-1
subnet-********* (10.0.65.0/24) | us-east-1b analytics-dev-db-subnet-2
subnet-********* (10.0.32.0/19) | us-east-1b analytics-dev-subnet-2
subnet-********* (10.0.0.0/19) | us-east-1a analytics-dev-subnet-1

Security Groups
sg-********* (sls-data-analytics-dev-r-AnalyticsLambdaEc2SecurityGroup-*********) | analytics-dev-lambda-security-groupsg-********* (sls-data-analytics-dev-r-AnalyticsDbEc2SecurityGroup-*********) | analytics-dev-db-security-groupsg-********* (default)

Is the serverless framework really adding all of these subnets and security groups or is AWS reporting on the associations in a strange way?  Is there an alternate method of verifying that the associations are correct?

When my lambda executed it did take an IP from one of the "compute subnets" and not a "db subnet", but I don't know if that was just a coincidence.

**Comment by mikelax** [comment]
I have a a similar situation where I have a Lambda that needs to connect to an RDS database in a VPC. In my advanced settings for the Lambda, I see the entire list of security groups for my VPC, about 10. In that list, only the one security group I defined in s-function is actually selected though. 

So I think sls is configuring correctly, this is just a display issue with the AWS lambda console.

**Comment by hypexr** [comment]
@mikelax How do you know it's a display issue?  Is there a way to verify?  I tried all of the aws cli options without any luck.

**Comment by mikelax** [comment]
@hypexr For my lambda function, in the s-function vpc section I have one security group and two subnet ids configured. 

When I run the following on the aws cli I see correctly one group and two subnets listed
`aws lambda get-function --function-name my-function-name`
I am using the latest version of the aws cli
```
Mike@Michaels-MBP core (master) $ aws --version
aws-cli/1.10.33 Python/2.7.10 Darwin/15.4.0 botocore/1.4.23
```

When I look at the advanced settings of the same function in the AWS console, I see all subnets listed, while only the two I defined in `s-function` are selected. Similarly, in security groups I see listed about 10 groups listed, but again only the one I defined in `s-function` is selected.
